apm-sniffer/apm-sdk-plugin/dubbo-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/asf/dubbo3/DubboInterceptor.java

```diff
@@ -65,15 +63,7 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         Invoker invoker = (Invoker) allArguments[0];
         Invocation invocation = (Invocation) allArguments[1];
 
-        RpcServiceContext serviceContext = RpcContext.getServiceContext();
-        boolean isConsumer;
-        URL url = serviceContext.getUrl();
-        if (url == null) {
-            url = serviceContext.getConsumerUrl();
-            isConsumer = url != null;
-        } else {
-            isConsumer = serviceContext.isConsumerSide();
-        }
+        boolean isConsumer = !isProvider(invocation);
```

Comment:
Is better to change method name to isConsumer? Assigning Not isProvider to a variable seems strange.

Reply:
Agree